pkg/virt-controller/watch/migration.go

```diff
@@ -422,6 +426,20 @@ func (c *MigrationController) sync(migration *virtv1.VirtualMachineInstanceMigra
 			return c.createTargetPod(migration, vmi)
 		}
 		return nil
+	case virtv1.MigrationAborting:
+		vmiCopy := vmi.DeepCopy()
+		if vmiCopy.Status.MigrationState != nil {
+			vmiCopy.Status.MigrationState.Aborted = true
+			if !reflect.DeepEqual(vmi.Status, vmiCopy.Status) ||
+				!reflect.DeepEqual(vmi.Labels, vmiCopy.Labels) {
+				_, err := c.clientset.VirtualMachineInstance(vmi.Namespace).Update(vmiCopy)
```

Comment:
Since aborting the migration involves deleting the migration object which in turn needs to perform some action on the VMI, I think the migration object needs a finalizer.

Without a finalizer, it's possible for the migration object to be deleted and removed from the cluster without this update ever being successful.

For example, if this update fails it's possible for the migration object to get removed from the cluster without the work queue ever retrying.

Reply:
Yeah we definitely need a finalizer! :+1: